File: VB/CustomActionControl.Module/Order.vb
Imports System
Imports System.Linq
Imports System.Text
Imports DevExpress.Xpo
Imports DevExpress.ExpressApp
Imports System.ComponentModel
Imports DevExpress.ExpressApp.DC
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.Base
Imports System.Collections.Generic
Imports DevExpress.ExpressApp.Model
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace CustomActionControl.Module
    <DefaultClassOptions> _
    Public Class Order
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        Private _OrderNo As String
        Public Property OrderNo() As String
            Get
                Return _OrderNo
            End Get
            Set(ByVal value As String)
                SetPropertyValue("OrderNo", _OrderNo, value)
            End Set
        End Property
        Private _Active As Boolean
        Public Property Active() As Boolean
            Get
                Return _Active
            End Get
            Set(ByVal value As Boolean)
                SetPropertyValue("Active", _Active, value)
            End Set
        End Property
        Private _Date As Date
        Public Property [Date]() As Date
            Get
                Return _Date
            End Get
            Set(ByVal value As Date)
                SetPropertyValue("Date", _Date, value)
            End Set
        End Property
    End Class
End Namespace

File: CS/CustomActionControl.Module/Order.cs
using System;
using System.Linq;
using System.Text;
using DevExpress.Xpo;
using DevExpress.ExpressApp;
using System.ComponentModel;
using DevExpress.ExpressApp.DC;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using System.Collections.Generic;
using DevExpress.ExpressApp.Model;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace CustomActionControl.Module {
    [DefaultClassOptions]
    public class Order : BaseObject {
        public Order(Session session)
            : base(session) {
        }
        private string _OrderNo;
        public string OrderNo {
            get { return _OrderNo; }
            set { SetPropertyValue("OrderNo", ref _OrderNo, value); }
        }
        private bool _Active;
        public bool Active {
            get { return _Active; }
            set { SetPropertyValue("Active", ref _Active, value); }
        }
        private DateTime _Date;
        public DateTime Date {
            get { return _Date; }
            set { SetPropertyValue("Date", ref _Date, value); }
        }
    }
}
